src/Tinfoil/Data/Key.hs

```diff
@@ -62,3 +72,8 @@ instance Eq (SecretKey a) where
 
 instance NFData (SecretKey a) where
   rnf (SKey_Ed25519 x) = rnf x
+
+instance Show (SecretKey a) where
+  showsPrec p (SKey_Ed25519 _) =
+    showParen (p > appPrec) $
+      showString "SecretKey SKey_Ed25519" . showsPrec appPrec1 ("redacted" :: ByteString)
```

Comment:
Shouldn't this render as `SKey_Ed25519 "redacted"` ? (rather than `SecretKey SKey_Ed25519 "redacted"`)